src/iuse.cpp

```diff
@@ -6411,20 +6487,769 @@ struct npc_photo_def : public JsonDeserializer, public JsonSerializer {
     }
 };
 
+const auto trap_name_at = []( const tripoint &point )
+{
+    auto trap = g->m.tr_at( point );
+    std::string name;
+    if( !trap.is_null() && trap.get_visibility() <= 1 ) {
+        name = colorize( trap.name(), trap.color ) + _( " on " );
+    };
+    return name;
+};
+
+std::string get_field_description_at( const tripoint &point )
+{
+    typedef std::unordered_set<field_id, std::hash<int>> Field_id_set;
+    static const Field_id_set covered_in_affix_ids = {
```

Comment:
I assumed that my solution will improve readability and reduced lines of code, since it is data structure really

Reply:
May be i should make next PR with jsonization of fields and remove these dictionaries?

Reply:
>I assumed that my solution will improve readability and reduced lines of code, since it is data structure really

I mostly suggested this because it wouldn't trigger that compiler error you mentioned (but I guess I failed to mention it :P).  A switch should get optimized to a hash table by the compiler anyways, so no worries there.  I do like your data structure here, but I don't think the massive switch would be much, just take up way more lines of code.

>May be i should make next PR with jsonization of fields and remove these dictionaries?

That would be super awesome, fields are one of the few major things that are still 100% hardcoded in c++.